wayat/frontend/lib/features/profile/pages/preferences_page/preferences_page.dart

```diff
@@ -3,17 +3,24 @@ import 'package:flutter_mobx/flutter_mobx.dart';
 import 'package:get_it/get_it.dart';
 import 'package:restart_app/restart_app.dart';
 import 'package:wayat/app_state/profile_state/profile_state.dart';
+import 'package:wayat/common/theme/colors.dart';
 import 'package:wayat/common/widgets/switch.dart';
 import 'package:wayat/features/profile/controllers/edit_profile_controller.dart';
 import 'package:wayat/lang/app_localizations.dart';
 import 'package:wayat/lang/lang_singleton.dart';
 import 'package:wayat/lang/language.dart';
 
-class PreferencesPage extends StatelessWidget {
-  PreferencesPage({Key? key}) : super(key: key);
+class PreferencesPage extends StatefulWidget {
+  const PreferencesPage({super.key});
 
+  @override
+  State<PreferencesPage> createState() => _PreferencesPageState();
+}
+
+class _PreferencesPageState extends State<PreferencesPage> {
   final EditProfileController controller = EditProfileController();
   final ProfileState profileState = GetIt.I.get<ProfileState>();
+  Language? changedLanguage;
```

Comment:
It should be as default value, the value used in the dropdown button. Doing this it should not restart the app if the user selects the same language as before.

Reply:
fix